nibabel/filebasedimages.py

```diff
@@ -18,6 +20,10 @@
 from .filename_parser import TypesFilenamesError, splitext_addext, types_filenames
 from .openers import ImageOpener
 
+FileSpec = ty.Union[str, os.PathLike]
```

Comment:
Since we're already going for 3.10 style annotation, we might as well:
```suggestion
FileSpec = str | os.PathLike
```
Shouldn't we?

Reply:
This line gets run by the interpreter, so fails before Python 3.10. I can try hiding it behind `if ty.TYPE_CHECKING:`. but then I can't import `FileSpec` into other modules without guarding that behind `ty.TYPE_CHECKING` either.

Reply:
I see. Thank you for clarifying. Definitely sounds like keeping the `Union` makes more sense in this case.